agent/engine/engine_sudo_linux_integ_test.go

```diff
@@ -238,6 +262,31 @@ func TestFirelensFluentbit(t *testing.T) {
 	assert.Error(t, err)
 }
 
+// getPortFromAddr returns the port part of an address in format "http://<addr>:<port>".
+func getPortFromAddr(addr string, t *testing.T) string {
+	parts := strings.Split(addr, ":")
+	if len(parts) < 2 {
+		t.Fatalf("Unable to get port from address. Invalid address: %s", addr)
+	}
+	return parts[len(parts)-1]
+}
+
+// getHostPrivateIP returns the host's private IP.
+func getHostPrivateIP(ec2MetadataClient ec2.EC2MetadataClient, t *testing.T) string {
+	ip, err := ec2MetadataClient.PrivateIPv4Address()
+	require.NoError(t, err)
+	return ip
+}
+
+// setV3MetadataURLFormat sets the container metadata URI format and returns a function to set it back.
+func setV3MetadataURLFormat(fmt string) func() {
+	oFmt := apicontainer.MetadataURIFormat
```

Comment:
nit: can we call this variable backup so that it's more meaningful what the purpose of this var is?

Reply:
updated